code/game/objects/items/eightball.dm

```diff
@@ -41,6 +41,13 @@
 		new /obj/item/toy/eightball/haunted(get_turf(src))
 		qdel(src)
 
+/obj/item/toy/eightball/examine(mob/user)
+	..()
+	if(on_cooldown)
```

Comment:
@coiax it has to be possible, how do you think the timing subsystem does it? it's similar to qdelling too:
1. Add a timer set for 30 seconds in the future
2. Label that time as happening at ```world.time + 300ds```
3. Continually loop the list every process to see if any are ready, and if they are fire them off.

Even if it's not immediately possible, all it'd take is adding that future time onto the timer callback itself (it'd have to be a callback datum), and you'd have to find that callback datum by storing the timer in a variable (to get its ID)

tl;dr yes.